### PR TITLE
Add more Vue.js recipes

### PR DESCRIPTION
**Pull request by suchmaske**
I have struggled the last days to properly test vue components using ava. To make life easier for others, I added some additional recipes for testing vuejs using ava. 

For example:

* testing the DOM of the component after a change in the data model
* testing data model changes when using async requests and promises
* testing DOM updates when promises are involved

### CODE REVIEW

File: docs/recipes/vue.md [hunk 1]
```diff
@@ -60,6 +80,71 @@ test('renders', t => {
...
+import Component from 'component.vue';
+
+test('see the updated message', t => {
+	const vm = new Vue(Component).$mount();
+	t.is(vm.$el.textContent, 'my message');
+
+	vm.message = 'my new message';
+	// this fails here: t.is('my new message', vm.$el.textContent)
+
+	Vue.nextTick(() => {
```
Comment (novemberborn) [inline docs/recipes/vue.md]: I assume this is asynchronous? That'll mean the assertion is run when the test is ended. We don't currently warn about that, but it means the assertion is ignored. You'd have to get a promise out of this or use `test.cb()` with `t.end()`.

Comment (sindresorhus) [inline docs/recipes/vue.md]: From the Vue docs:

> New in 2.1.0+: returns a Promise if no callback is provided and Promise is supported in the execution environment. - https://vuejs.org/v2/api/#Vue-nextTick

So you can just use an async function for the test and `await` it here.

File: docs/recipes/vue.md [hunk 2]
```diff
@@ -60,6 +80,71 @@ test('renders', t => {
...
+
+// Don't forget to add the async prefix
+test('update the model using promises', async t => {
+	// assume the message data property is a pending promise defined in the Component.
+	const vm = new Vue(Component).$mount();
+
+	// just testing the data model can be achieved by using await
+	t.is(await vm.message, 'my message');
+
+	await vm.message.then((data) => {
```
Comment (sindresorhus) [inline docs/recipes/vue.md]: The `.then()` here is moot. You're already `await`'ing it. Just get value back the async/await way.

File: docs/recipes/vue.md [hunk 3]
```diff
@@ -36,9 +38,27 @@ const Vue = require('vue');
 // Setup Vue.js to remove production tip
 Vue.config.productionTip = false;
 
+// Setup Vue files to be processed by `require-extension-hooks-vue`
+hooks('vue').plugin('vue').push();
+// Setup Vue and JS files to be processed by `require-extension-hooks-babel`
+hooks(['vue', 'js']).plugin('babel').push();
+```
+
+When you are using ES6 modules, use the dist-file of Vue. Not doing that may cause a not completely accessible (and thus not testable) Vue model.
```
Comment (sindresorhus) [inline docs/recipes/vue.md]: ES6 => ES2015

File: docs/recipes/vue.md [hunk 4]
```diff
@@ -36,9 +38,27 @@ const Vue = require('vue');
...
+
+```js
+import browserEnv from 'browser-env';
+import hooks from 'require-extension-hooks';
+import Vue from 'vue/dist/vue';
```
Comment (OmgImAlexis) [inline docs/recipes/vue.md]: Does this type of import not require webpack? I didn't think this worked out of the box with babel, etc. yet.

**Comment by sindresorhus** [comment]
@suchmaske @blake-newman @OmgImAlexis @knpwrs ping :)

File: docs/recipes/vue.md (hunk 3 above)
Comment (blake-newman) [reply docs/recipes/vue.md]: This is not true. Using the hooks compiles the es6 to es2015. You can use the source components.

File: docs/recipes/vue.md (hunk 4 above)
Comment (blake-newman) [reply docs/recipes/vue.md]: Not sure of the point of this example, same as above but es imports. 

@sindresorhus can you confirm @OmgImAlexis is also correct.

File: docs/recipes/vue.md (hunk 1 above)
Comment (blake-newman) [reply docs/recipes/vue.md]: Agree with using async and await syntax.

File: docs/recipes/vue.md [hunk 5]
```diff
@@ -60,6 +80,71 @@ test('renders', t => {
...
+import Component from 'component.vue';
+
+// Don't forget to add the async prefix
+test('update the model using promises', async t => {
+	// assume the message data property is a pending promise defined in the Component.
+	const vm = new Vue(Component).$mount();
+
+	// just testing the data model can be achieved by using await
+	t.is(await vm.message, 'my message');
+
```
Comment (blake-newman) [inline docs/recipes/vue.md]: Why would vm.message be a promise? No need to `await` this

**Comment by knpwrs** [comment]
I think it's worth pointing out that [Vue has started work on official testing tools](https://github.com/vuejs/vue-test-utils). They have an [example with AVA](https://github.com/eddyerburgh/vue-test-utils-ava-example) available.

**Comment by blake-newman** [comment]
@knpwrs Indeed. Will create a PR to link to that example repository

**Comment by novemberborn** [comment]
@blake-newman @knpwrs is this PR still relevant?

@suchmaske will you have time to revive this PR?

**Comment by suchmaske** [comment]
@novemberborn I recently have changed my workflow to [vue-test-utils](https://github.com/vuejs/vue-test-utils). I will take the feedback and try to rework the PR in the next days.

**Comment by knpwrs** [comment]
I don't personally use Vue enough anymore to offer any official recommendations.

**Comment by novemberborn** [comment]
That's great, thank you @suchmaske.

**Comment by blake-newman** [comment]
@novemberborn i don't think the PR is relevant. Also there is many examples, via `vue-test-utils` that enhance what is on offering here. 

Sorry for the late reply

**Comment by sindresorhus** [comment]
Closing this for lack of activity. Happy to reopen if anyone wants to finish it.

**Pull request closed without merging**